core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -396,12 +396,12 @@ class DAGScheduler(
 
   /** Find ancestor shuffle dependencies that are not registered in shuffleToMapStage yet */
   private def getMissingAncestorShuffleDependencies(
-      rdd: RDD[_]): Stack[ShuffleDependency[_, _, _]] = {
-    val ancestors = new Stack[ShuffleDependency[_, _, _]]
+      rdd: RDD[_]): ArrayStack[ShuffleDependency[_, _, _]] = {
```

Comment:
Stack is deprecated in 2.12 for poor performance; ArrayStack should work the same and be faster, in 2.11 too